compiler/rustc_resolve/src/build_reduced_graph.rs

```diff
@@ -169,6 +169,15 @@ impl<'ra, 'tcx> Resolver<'ra, 'tcx> {
         }
     }
 
+    /// Add every proc macro accessible from the current crate to the `macro_map` so diagnostics can
+    /// find them for suggestions.
+    pub(crate) fn register_macros_for_all_crates(&mut self) {
+        let def_ids = self.cstore().all_proc_macro_def_ids();
+        for def_id in def_ids {
+            self.get_macro_by_def_id(def_id);
+        }
+    }
```

Comment:
We should memoize this, but the `Resolver` doesn't use queries, so we should likely do that "manually" by adding a signal field to the `Resolver` to skip this logic after the first call.